### PR TITLE
Make csrank tf2 api-compatible

### PR DESCRIPTION
**Pull request by timokau**
## Description

Using the upgrade script: https://www.tensorflow.org/guide/upgrade
Worked around https://github.com/google/pasta/issues/79 by first
doing

rg -l 'metaclass' | while read file; do sed -i 's/metaclass=/metaclassEQUALS/g' "$file"; done

Then running the update script and finally running the reverse sed script.

As mentioned in the above link, this does not necessarily create
idiomatic tf2 code. It does provide api-compatibility with tf2 however.
Since tf 1.14 and 1.15 are forwards compatible, we remain compatible
with tf1 as well.

The update script reported no issues.

Future work will be testing cs-ranking with tf2 (for now I tested with
1.15) and enabling tf2 behaviour.

## Motivation and Context

Tensorflow 2 introduced many breaking changes. 1.15 is the last release that supports the old api.

## How Has This Been Tested?

Ran all tests with tf 1.15.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by timokau** [comment]
I've pushed a WIP commit that is supposed to make cs-ranking actually compatible with v2 behaviour. Most changes are straightforward, but the change to eager execution is a big one. It needs some significant changes in our learners, since graph buildup and model execution are no longer separate. We need to remove all usages of `tf.Session` and `tf.placeholder`.

Do we want to go down this path? Alternatives would be

1. keep cs-ranking v1.1 dependent on tf1
2. explicitly disable v2 behaviour, which is essentially equal to the first option (since we will force all consumers of the library to use v1 behaviour as well)
3. port to pytorch instead

Note that a tf2 port would probably also make an eventual pytorch port simpler, since pytorch executes eagerly as well.

**Comment by timokau** [comment]
Private discussion with @kiudee resulted in option (1), since (3) is too much effort for version 1.1 and (2) is not easily feasible and has little advantage.
